docs/antrea-network-policy.md

```diff
@@ -365,19 +412,34 @@ Usage of ClusterGroups along with stand-alone selectors is not allowed.
 
 ### Behavior of *to* and *from* selectors
 
-There are four kinds of selectors that can be specified in an ingress `from`
+There are six kinds of selectors that can be specified in an ingress `from`
 section or egress `to` section:
 
 **podSelector**: This selects particular Pods from all Namespaces as "sources",
 if set in `ingress` section, or as "destinations", if set in `egress` section.
 
 **namespaceSelector**: This selects particular Namespaces for which all Pods
-are grouped as `ingress` "sources" or `egress` "destinations".
+are grouped as `ingress` "sources" or `egress` "destinations". Cannot be set
+with `namespaces` field.
 
 **podSelector** and **namespaceSelector**:  A single to/from entry that
 specifies both namespaceSelector and podSelector selects particular Pods within
 particular Namespaces.
 
+**namespaces**: A `namespaces` field allows users to perform advanced matching on
+Namespace objects which cannot be done via label selectors. Currently, the
+`namespaces` field has only one matching strategy, `self`. If set to `self`, it indicates
+that the corresponding `podSelector` (or all Pods if `podSelector` is not set)
+should only select Pods belonging to the same Namespace as the workload targeted
+(either through a policy-level AppliedTo or a rule-level Applied-To) by the current
```

Comment:
I feel like there is a lot to unpack from this sentence even though it is as clear as I can think of... maybe a diagram will help in the future.

Reply:
i think since we plan to add more strategies, it may make sense to devote a sub section for Namespaces; where in we can do further explanation and diagrams for all new types